design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,79 @@
+# Queries
+
+```graphql
+type Query {
+    multiSearch(phrase: String!, productSize: Int = 10): MultiSearchResponse!
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(phrase: String!, filter: [SearchClauseInput], sort: [ProductSearchSortInput]): ProductSearchResponse!
+}
+
+input ProductSearchSortInput
+{
+    attribute: String!
+    direction: SortEnum!
+}
+
+# If from or to fields are omitted, $gte or $lte filter will be applied
+input SearchRangeInput {
+    from: Int
+    to: Int
+}
+
+input SearchClauseInput {
+    attribute: String!
+    in: [String]
+    eq: String
+    range: SearchRangeInput
+}
+
+type ProductSearchResponse {
+    items: [ProductSearchItem]
+    hits: Int!
+    facets: [Aggregation]
+    facetsValues: [Aggregation]
+}
```

Comment:
for search response, it will probably make sense to have a request phrase as well.
Especially if we would support "Did You Mean?" functionality where we would automatically change the initial query, for example, fixing smelling there. 

Also, It might make sense to show the "real query" which was used for search.
For example, "IPhane" search request to google - https://www.google.com/search?q=iphane&rlz=1C5CHFA_enUA813UA813&oq=iphane&aqs=chrome..69i57j0l7.4027j0j8&sourceid=chrome&ie=UTF-8

Makes automatic search for corrected version "iphone", but along with that provides possibility to make a search by "iphane"

```
Showing results for iphone
Search instead for iphane
```
And if a user wants to see results for `IPhane` we also need to specify a way to not modify initial search phrase

Reply:
Added suggestions to the response

Reply:
I think we don't have search phrase auto swap today, so skipping this points. We can add it later in search response